### ISSUE TITLE
[Bug] when sglang received over 16 concurrency request(means i create 16 thread to call the service all the time), it will return abnormal result, and in the log, will occur NaN 

### ISSUE THREAD

**Issue by GGBond8488** [sgl-project/sglang#1742]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [X] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 5. Please use English, otherwise it will be closed.

### Describe the bug

the number below means there are 8/16/24 threads continuously request the sglang service in the same time. 

```
########6#########
Average input tokens num: 838
Average generate tokens num: 95

########8#########
Average input tokens num: 838
Average generate tokens num: 95

########16#########
Average input tokens num: 838
Average generate tokens num: 167

########24#########
Average input tokens num: 838
Average generate tokens num: 181
```

we can see that, the output tokens are not stable when concurrency >=16
and i notice that there are many abnormal logs like these:

```
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Detected errors during sampling! NaN in the probability.
[00:04:06 TP0] Prefill batch. #new-seq: 1, #new-token: 757, #cached-token: 0, cache hit rate: 0.00%, token usage: 0.06, #running-req: 23, #queue-req: 0
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
INFO:     127.0.0.1:41288 - "POST /generate HTTP/1.1" 200 OK
[00:04:07 TP0] Detected errors during sampling! NaN in the probability.
```

BTW, I found sglang doesn’t fully use the computing power of the GPU：
<img width="773" alt="image" src="https://github.com/user-attachments/assets/174324d4-a587-406e-b3f3-1dca952473ec">

### Reproduction

run_script:
```
python3 -m sglang.launch_server \
	--model-path $MODEL_PATH \
	--tp ${TP} \
	--disable-radix-cache \
	--host 0.0.0.0 \
	--port ${PORT} \
	--log-level debug \
	--file-storage-pth /code/llama31.storage \
	--chunked-prefill-size -1 \
	--mem-fraction-static ${mem_fraction_static} \
	--quantization fp8 \
```

set T1=1, MODEL_PATH=llama3.1-8b, mem_fraction_static=0.75 

test_script:
```python
def make_api_call(REQUEST_PAYLOAD):
    try:
        start_time = time.time()
        headers = {"Content-Type": "application/json"}
        response = requests.post(API_ENDPOINT, headers=headers, data=json.dumps(REQUEST_PAYLOAD))
        response.raise_for_status()
        end_time = time.time()
        request_time = end_time - start_time
        
    except requests.exceptions.RequestException as e:
        print(e, flush=True)

with concurrent.futures.ThreadPoolExecutor(max_workers=CONCURRENT_REQUESTS) as executor:
    # Submit API call tasks
    results = [executor.submit(make_api_call, REQUEST_PAYLOAD) for _, REQUEST_PAYLOAD in enumerate(REQUEST_PAYLOADS)]

    # Wait for all tasks to complete
    concurrent.futures.wait(results)
```

### Environment

sglang
```
Name: sglang
Version: 0.3.3.post1
Summary: SGLang is yet another fast serving framework for large language models and vision language models.
Home-page: https://github.com/sgl-project/sglang
Author: 
Author-email: 
License: 
Location: /usr/local/lib/python3.10/dist-packages
Editable project location: /sgl-workspace/sglang/python
Requires: numpy, requests, tqdm
```

cuda：
<img width="773" alt="image" src="https://github.com/user-attachments/assets/3386cef8-1714-4ba5-a4d5-480c7215320f">

os:
```
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by merrymercy** [comment]
1. Can you try `--disable-cuda-graph-padding`?
2. Does it work correctly when the concurrency is low?
3. What if you run with bf16, i.e., drop `--quant fp8`.

**Comment by 81549361** [comment]
> 1. 你能嘗試嗎`--disable-cuda-graph-padding`？
> 2. 并发量较低时能正常工作吗？
> 3. 如果使用 bf16 运行，即 drop `--quant fp8`，会怎么样呢？
除了并发较低时正常工作，其他两个选项依然出现此错误。

**Comment by merrymercy** [comment]
@GGBond8488 @81549361 
Any reproducible scripts will be helpful for us!

**Comment by merrymercy** [comment]
fixed by #1779

**Comment by beepoos** [comment]
Hello sir！I would like to ask another question about multi-concurrency. I would like to ask whether the multi-concurrency in your script is effective and whether the request time is greatly reduced. I conducted a multi-concurrency test and the results show that There is no multi-concurrency effect. In fact, it is still serial processing.
